lib/google/youtube.js

```diff
@@ -1,18 +1,35 @@
-// const { DateTime } = require('luxon')
 const { authenticate } = require('./authenticate')
 const { google } = require('googleapis')
 
-const getStreams = async ({ max = 5, status = 'upcoming' } = {}) => {
+const getStreamsData = async ({ max = 5, status = 'upcoming', pageToken = '' } = {}) => {
   const auth = authenticate('youtube')
   const youtube = google.youtube({ version: 'v3', auth })
 
   const streams = await youtube.liveBroadcasts.list({
     broadcastStatus: status,
     maxResults: max,
-    part: 'id,snippet,contentDetails,status'
+    part: 'id,snippet,contentDetails,status',
+    pageToken
   })
 
-  return streams.data.items || []
+  return streams.data || {}
 }
 
-module.exports = { getStreams }
+const getStreams = async ({ max = 5, status = 'upcoming', pageToken = '' } = {}) => {
+  const streamsData = await getStreamsData({ max, status, pageToken })
+
+  return streamsData.items || []
+}
+
+const getLastStream = async ({ max = 1, status = 'upcoming', pageToken = '' } = {}) => {
+  const streams = await getStreamsData({ max, status, pageToken })
```

Comment:
you misunderstood,
I suggest to set `max = 1` just for testing, we should NOT force the script to make so many requests.

Set it to `100` (or whatever is the max allowed from youtube) for the final version

Reply:
The `getLastStream` does not need to accept the neither `max` nor the `status` arguments, as its name suggest it will always reach until the last one,

```suggestion
const MAX_STREAMS = 100; //check the best limit to be used and move this to before function definitions

const getLastStream = async ({ pageToken = '' } = {}) => {
  const streams = await getStreamsData({ max: MAX_STREAMS, status: 'upcoming', pageToken })
```

Reply:
Done!